retrofit-converters/moshi/src/main/java/retrofit2/converter/moshi/MoshiConverterFactory.java

```diff
@@ -40,20 +40,36 @@ public static MoshiConverterFactory create() {
 
   /** Create an instance using {@code moshi} for conversion. */
   public static MoshiConverterFactory create(Moshi moshi) {
-    return new MoshiConverterFactory(moshi);
+    return create(moshi, false);
+  }
+
+  private static MoshiConverterFactory create(Moshi moshi, boolean lenient) {
+    return new MoshiConverterFactory(moshi, lenient);
   }
 
   private final Moshi moshi;
+  private final boolean lenient;
 
-  private MoshiConverterFactory(Moshi moshi) {
+  private MoshiConverterFactory(Moshi moshi, boolean lenient) {
     if (moshi == null) throw new NullPointerException("moshi == null");
     this.moshi = moshi;
+    this.lenient = lenient;
+  }
+
+  public MoshiConverterFactory asLenient() {
+    if (lenient) {
```

Comment:
I would just delete this `if`. It'll never happen and even if it does it's not that big of a deal to create a new instance.